File: 535-encode-and-decode-tinyurl/535-encode-and-decode-tinyurl.cpp
class Solution {
public:

    unordered_map<string,string> umap;
    int n = 0;
    // Encodes a URL to a shortened URL.
    string encode(string longUrl) {
        string res = "http://tinyurl.com/" + to_string(n);
        n++;
        umap[res]=longUrl;
        return res;
    }

    // Decodes a shortened URL to its original URL.
    string decode(string shortUrl) {
        return umap[shortUrl];
    }
};

// Your Solution object will be instantiated and called as such:
// Solution solution;
// solution.decode(solution.encode(url));

File: 535-encode-and-decode-tinyurl/535-encode-and-decode-tinyurl.java
public class Codec {

    Map<String,String> mp = new HashMap<>();
    int n =0;
    // Encodes a URL to a shortened URL.
    public String encode(String longUrl) {
        String res = "https://leetcode.com/problems/" + Integer.toString(n);
        mp.put(res, longUrl);
        return res;
    }

    // Decodes a shortened URL to its original URL.
    public String decode(String shortUrl) {
        return mp.get(shortUrl);
    }
}

// Your Codec object will be instantiated and called as such:
// Codec codec = new Codec();
// codec.decode(codec.encode(url));